### ISSUE TITLE
Unable to download file by method FOLDER if another driver in the using block was used before

### ISSUE THREAD

**Issue by ilya-corp** [selenide/selenide#2202]
Open your site and perform some actions:
`
Selenide.open("http://your.site.com");`

Create new driver and open another site using this driver using block "using":
```
ChromeDriver customDriver = new ChromeDriver();
using(customDriver, () -> {
                open("http://another.site", AuthenticationType.BASIC, new BasicAuthCredentials("login", "password"));
            });
customDriver.close();
```

After that download by default driver something using method FOLDER:
 `File csvReportFile = historyPage.downloadCSVButton.getInstance().download(DownloadOptions.using(FileDownloadMode.FOLDER).withFilter(FileFilters.withExtension("csv")).withTimeout(Duration.ofSeconds(15L)));
`
As result Selenide can't find file with extension "CSV" in the downloads folder. But really file exists.
If I change using method of custom driver like following
`customdriver.find(By.cssSelector("div.super")).click();`
then everything works as expected.

Please see related issue https://github.com/selenide/selenide/issues/2186

## Tell us about your environment
* **Selenide Version**: 6.12.2
* **Chrome\Firefox\IE Version**: Chrome 110
* **Browser Driver Version**: 
* **Selenium Version**: 
* **OS Version**: Windows 10

## Code To Reproduce Issue [ Good To Have ]
Please remember that with sample code it's easier to reproduce the bug.

**Comment by asolntsev** [comment]
@ilya-corp Good catch! It seems proxy was also broken after using `using`. :) 
Will be fixed in Selenide 6.12.3

**Comment by ilya-corp** [comment]
Thank you, Andrey, for the fixing this issue!
Have a good day!
